recipes/metis/all/patches/002-support-pure-cmake.patch

```diff
@@ -0,0 +1,771 @@
+-  METIS_ERROR_MEMORY    = -3,   /*!< Returned due to insufficient memory */
+-  METIS_ERROR           = -4    /*!< Some other errors */
+-} rstatus_et; 
+-
+-
+-/*! Operation type codes */
+-typedef enum {
+-  METIS_OP_PMETIS,       
+-  METIS_OP_KMETIS,
+-  METIS_OP_OMETIS
+-} moptype_et;
+-
+-
+-/*! Options codes (i.e., options[]) */
+-typedef enum {
+-  METIS_OPTION_PTYPE,
+-  METIS_OPTION_OBJTYPE,
+-  METIS_OPTION_CTYPE,
+-  METIS_OPTION_IPTYPE,
+-  METIS_OPTION_RTYPE,
+-  METIS_OPTION_DBGLVL,
+-  METIS_OPTION_NIPARTS,
+-  METIS_OPTION_NITER,
+-  METIS_OPTION_NCUTS,
+-  METIS_OPTION_SEED,
+-  METIS_OPTION_NO2HOP,
+-  METIS_OPTION_ONDISK,
+-  METIS_OPTION_MINCONN,
+-  METIS_OPTION_CONTIG,
+-  METIS_OPTION_COMPRESS,
+-  METIS_OPTION_CCORDER,
+-  METIS_OPTION_PFACTOR,
+-  METIS_OPTION_NSEPS,
+-  METIS_OPTION_UFACTOR,
+-  METIS_OPTION_NUMBERING,
+-  METIS_OPTION_DROPEDGES,
+-
+-  /* Used for command-line parameter purposes */
+-  METIS_OPTION_HELP,
+-  METIS_OPTION_TPWGTS,
+-  METIS_OPTION_NCOMMON,
+-  METIS_OPTION_NOOUTPUT,
+-  METIS_OPTION_BALANCE,
+-  METIS_OPTION_GTYPE,
+-  METIS_OPTION_UBVEC
+-} moptions_et;
+-
+-
+-/*! Partitioning Schemes */
+-typedef enum {
+-  METIS_PTYPE_RB, 
+-  METIS_PTYPE_KWAY                
+-} mptype_et;
+-
+-/*! Graph types for meshes */
+-typedef enum {
+-  METIS_GTYPE_DUAL,
+-  METIS_GTYPE_NODAL               
+-} mgtype_et;
+-
+-/*! Coarsening Schemes */
+-typedef enum {
+-  METIS_CTYPE_RM,
+-  METIS_CTYPE_SHEM
+-} mctype_et;
+-
+-/*! Initial partitioning schemes */
+-typedef enum {
+-  METIS_IPTYPE_GROW,
+-  METIS_IPTYPE_RANDOM,
+-  METIS_IPTYPE_EDGE,
+-  METIS_IPTYPE_NODE,
+-  METIS_IPTYPE_METISRB
+-} miptype_et;
+-
+-
+-/*! Refinement schemes */
+-typedef enum {
+-  METIS_RTYPE_FM,
+-  METIS_RTYPE_GREEDY,
+-  METIS_RTYPE_SEP2SIDED,
+-  METIS_RTYPE_SEP1SIDED
+-} mrtype_et;
+-
+-
+-/*! Debug Levels */
+-typedef enum {
+-  METIS_DBG_INFO       = 1,       /*!< Shows various diagnostic messages */
+-  METIS_DBG_TIME       = 2,       /*!< Perform timing analysis */
+-  METIS_DBG_COARSEN    = 4,	  /*!< Show the coarsening progress */
+-  METIS_DBG_REFINE     = 8,	  /*!< Show the refinement progress */
+-  METIS_DBG_IPART      = 16, 	  /*!< Show info on initial partitioning */
+-  METIS_DBG_MOVEINFO   = 32, 	  /*!< Show info on vertex moves during refinement */
+-  METIS_DBG_SEPINFO    = 64, 	  /*!< Show info on vertex moves during sep refinement */
+-  METIS_DBG_CONNINFO   = 128,     /*!< Show info on minimization of subdomain connectivity */
+-  METIS_DBG_CONTIGINFO = 256,     /*!< Show info on elimination of connected components */ 
+-  METIS_DBG_MEMORY     = 2048,    /*!< Show info related to wspace allocation */
+-} mdbglvl_et;
+-
+-
+-/* Types of objectives */
+-typedef enum {
+-  METIS_OBJTYPE_CUT,
+-  METIS_OBJTYPE_VOL,
+-  METIS_OBJTYPE_NODE
+-} mobjtype_et;
+-
+-
+-
+-#endif  /* _METIS_H_ */
+--- /dev/null
++++ b/include/metis.h.in
```

Comment:
This patch is really big 😱 I understand what it's doing but it hard to say if it's correct

Have you considered opening a PR upstream to contribute back? that can help us verify the work + not have to repeate this in the future 🤞

Reply:
It has been taken from closed PR
https://github.com/KarypisLab/METIS/pull/15/files

Does conan support rename in patches? With rename it will be smaller